### ISSUE TITLE
PackagesotFoundError on BertopicR::install_python_dependencies()

### ISSUE THREAD

**Issue by bumbygit** [jpcompartir/BertopicR#30]
When running the command `BertopicR::install_python_dependencies()` I get the error:

PackagesNotFoundError: The following packages are not available from current channels:

  - pytorch==2.0.0

Current channels:

  - https://conda.anaconda.org/conda-forge
  - defaults

I've verified that `torch version 2.0.0` is installed. Is there something I'm missing between `torch` and `pytorch` packages? The same error occurs when running the conda command `pip install pytorch==2.0.0` because it doesn't look like that version of `pytorch` exists

**Comment by jpcompartir** [comment]
Hi @bumbygit,

It used to be the case that conda handled torch installation via `conda install pytorch`, whereas pip was `pip install torch`, which is still the instruction on PyTorch's website: https://pytorch.org/get-started/locally/#mac-anaconda

However, recently they deprecated support for PyTorch nightly's on Conda - https://dev-discuss.pytorch.org/t/pytorch-deprecation-of-conda-nightly-builds/2590. 

I'm not sure if this is why the installer isn't finding pytorch via conda now - we'll investigate. 

Have you been able to use the rest of the package with your installation?

**Comment by bumbygit** [comment]
No, I haven't been able to use the rest of the package. But, to be fair, I haven't explored it fully and am trying to work through the tutorial at https://jpcompartir.github.io/BertopicR/. Without the ability to install dependencies, it seems to error out on the `bt_make_embedder()` function.

**Comment by jpcompartir** [comment]
I'll see if I can get around to checking/looking at a fix tomorrow - conveniently I have a new laptop to set up so I may have to fix it along the way. 

If you're feeling adventurous, it is possible to install the dependencies separately in the terminal - though it can be quite frustrating as we've frozen them and require particular versions + Rstudio's Python interpeter and terminal can be somewhat temperamental

**Comment by jpcompartir** [comment]
Following up here - 

Ran the installation on a new Macbook and torch etc. were all loaded appropriately.  But there is an issue with the installation of HF's hub -  think we probably need to version this in the python dependencies - sadly we have to version stuff precisely as we don't have the resources to keep up with the python ecosystem 😬 .

The README.md is actually out of date as we don't use bt_make_embedder anymore but moved to bt_make_embedder_* (st = sentence transformers, flair = flair etc.). So we will need to update this 👍 

Do you have the exact code you ran to install the environment, and did you use miniconda? 
Also, are you a mac, linux or windows user? I suspect there will be problems for the latter two as our team all run on macs so it's what we set the package up for and have tested.

**Comment by bumbygit** [comment]
Here's the output:

```
system("which python") # returns 127
py_config()
```

python:         C:/Users/.../Anaconda3/envs/BertopicR/python.exe
libpython:      C:/Users/.../Anaconda3/envs/BertopicR/python311.dll
pythonhome:     C:/Users/.../Anaconda3/envs/BertopicR
version:        3.11.10 | packaged by Anaconda, Inc. | (main, Oct  3 2024, 07:22:26) [MSC v.1929 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/.../Anaconda3/envs/BertopicR/Lib/site-packages/numpy
numpy_version:  2.0.2

```
'bertopic' %in% py_packages$package # FALSE
'sentence-transformers' %in% py_packages$package # FALSE
```

**Comment by jpcompartir** [comment]
To confirm, was this in a fresh R session, and do you have the following settings in Rstudio tools --> global options?

![image](https://github.com/user-attachments/assets/f3c26caf-2e13-4ac2-97a3-6f9efd061a23)

From what I can see, your packages have been set up using Anaconda (Anaconda3 in the path) rather than miniconda, the python version isn't syncing up with the expected, and numpy is also a higher version. All of these could cause problems. It suggests that something happened prior to the .onAttach(), or the steps somehow weren't executed and reticulate has defaulted to Anaconda.


Could you:
`reticulate::conda_list()` to see if we have a miniconda BertopicR as well as the Anaconda3?

If it were on my own machine, I would remove the current BertopicR environment completely, and start again - making sure along the way that everything was being done with miniconda rather than Anaconda.

That said, I expect there will be some other issues further along, so it may be less frustrating to wait until we get some development time to hotfix the issues.  if you are interested in persevering, it may be worth checking out [init.R](https://github.com/jpcompartir/BertopicR/blob/main/R/init.R) and manually running the environment set up (taking things out of the tryCatch).  

And then manually running the contents of install_python_dependencies() - admittedly this is not ideal but we've limited scope for figuring this out from our end without access to a Windows machine!

**Comment by bumbygit** [comment]
I did not have that setting in global options, but have made the change. I also did not see any miniconda BertopicR. I'll give another go at this tomorrow, but have a feeling it might end up waiting for your fixes because I may get out of my depth. I appreciate the help and diligence on this.
